pandas/tests/indexes/test_range.py

```diff
@@ -580,38 +580,62 @@ def test_union_noncomparable(self):
         expected = Index(np.concatenate((other, self.index)))
         tm.assert_index_equal(result, expected)
 
-    def test_union(self):
+    @pytest.fixture
+    def union_fixture(self):
+        """Inputs and expected outputs for RangeIndex.union tests"""
         RI = RangeIndex
         I64 = Int64Index
```

Comment:
so the to write this is like this

```
@pytest.fixture(params=[
    (RI(0, 10, 1), RI(0, 10, 1), RI(0, 10, 1), RI(0, 10, 1)),
    .......,
   ids=list_of_strings_describing_the_cases)
def unions(request):
      return request.param


def test_union(unions):
     idx1, idx2, expected_sorted, expected_nonsorted = unions
     ....
```

Reply:
Ok I understand what you mean now - thanks.